File: MachineProblem/Hard/Hard5.cpp
#include <bits/stdc++.h>
using namespace std;
int main(){
    map<string,int> cart;
    string cmd;
    while(cin >> cmd){
        if(cmd == "ADD"){
            string item; int price; cin >> item >> price; cart[item] = price;
        } else if(cmd == "REMOVE"){
            string item; cin >> item; cart.erase(item);
        } else if(cmd == "TOTAL"){
            long long total = 0;
            for(auto &p: cart) total += p.second;
            cout << "Total: " << total << endl;
        } else {
            // ignore unknown
        }
    }
    return 0;
}

File: MachineProblem/Hard/Hard5.java
import java.util.*;
public class Main {
    public static void main(String[] args){
        Scanner sc = new Scanner(System.in);
        Map<String,Integer> cart = new HashMap<>();
        while(sc.hasNext()){
            String cmd = sc.next();
            if(cmd.equals("ADD")){
                String item = sc.next(); int price = sc.nextInt(); cart.put(item, price);
            } else if(cmd.equals("REMOVE")){
                String item = sc.next(); cart.remove(item);
            } else if(cmd.equals("TOTAL")){
                long total = 0;
                for(int p: cart.values()) total += p;
                System.out.println("Total: " + total);
            } else {
                // ignore
            }
        }
    }
}
